fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java

```diff
@@ -31,11 +37,55 @@ protected ContentObject(PID pid, Repository repository, RepositoryObjectDataLoad
 		super(pid, repository, dataLoader);
 	}
 
-	public BinaryObject addDescription(InputStream contentStream) {
-		return null;
+	public FileObject addDescription(InputStream modsStream) {
```

Comment:
We are going to need a way to put new versions of the description in to support the edit description page.  My feel is that this should be done using the versioning api, but i haven't researched how that would work yet.  So given our current focus on the deposit application i think it's alright to just cover adding new descriptions, just something to keep in mind for later.